test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -319,3 +320,53 @@ func CheckCWSBehaviour(t *testing.T, client *TestClient) {
 		require.True(tt, result, "system-probe and security-agent should communicate")
 	})
 }
+
+// CheckSystemProbeBehavior runs tests to check the agent behave correctly when system-probe is enabled
+func CheckSystemProbeBehavior(t *testing.T, client *TestClient) {
+	t.Run("enable system-probe and restarts", func(tt *testing.T) {
+		err := client.SetConfig(client.Helper.GetConfigFolder()+"system-probe.yaml", "system_probe_config.enabled", "true")
+		require.NoError(tt, err)
+
+		err = client.SetConfig(client.Helper.GetConfigFolder()+"system-probe.yaml", "system_probe_config.enabled", "true")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Restart(client.Helper.GetServiceName())
+		require.NoError(tt, err, "datadog-agent should restart after CWS is enabled")
+	})
+
+	t.Run("system-probe is running", func(tt *testing.T) {
+		var err error
+		require.Eventually(tt, func() bool {
+			return AgentProcessIsRunning(client, "system-probe")
+		}, 1*time.Minute, 500*time.Millisecond, "system-probe should be running ", err)
```

Comment:
Have you benchmarked the start times? Is this sufficient?
Might be worth giving some extra room here to avoid unnecessary flakes in the CI

Reply:
I took the same times as used in the CWS tests, so I think that should work. Also they only check for the process running, not for complete boot of system-probe.